sync-core/src/main/java/com/cloudant/sync/query/QuerySqlTranslator.java

```diff
@@ -0,0 +1,386 @@
+
+        for (Object rawComponent: clause) {
+            Map<String, Object> component = (Map<String, Object>) rawComponent;
+            if (component.size() != 1) {
+                String msg = String.format("Expected single predicate per clause map, got %s",
+                                           component.toString());
+                logger.log(Level.SEVERE, msg);
+                return null;
+            }
+
+            String fieldName = (String) component.keySet().toArray()[0];
+            Map<String, Object> predicate = (Map<String, Object>) component.get(fieldName);
+
+            if (predicate.size() != 1) {
+                String msg = String.format("Expected single operator per predicate map, got %s",
+                                           predicate.toString());
+                logger.log(Level.SEVERE, msg);
+                return null;
+            }
+
+            String operator = (String) predicate.keySet().toArray()[0];
+
+            // $not specifies the opposite operator OR NULL documents be returned
+            if (operator.equals("not")) {
+                // TODO - implement NOT logic...
+            } else {
+                if (operator.equals(EXISTS)) {
+                    // TODO - implement EXISTS logic...
+                } else {
+                    String sqlOperator = operatorMap.get(operator);
+                    if (sqlOperator == null || sqlOperator.isEmpty()) {
+                        String msg = String.format("Unsupported comparison operator %s", operator);
+                        logger.log(Level.SEVERE, msg);
+                        return null;
+                    }
+
+                    String sqlClause = String.format("\"%s\" %s", fieldName, sqlOperator);
+                    sqlClause = addPredicateValueToSqlClause(sqlClause, predicate.get(operator));
+                    if (sqlClause == null) {
+                        logger.log(Level.SEVERE, "Unsupported predicate type");
+                        return null;
+                    }
+                    sqlClauses.add(sqlClause);
+                }
+            }
+
+        }
+        Joiner joiner = Joiner.on(" AND ").skipNulls();
+
+        return joiner.join(sqlClauses);
+    }
+
+    private static String addPredicateValueToSqlClause(String clausePrefix, Object predicateValue) {
+        String sqlClause = null;
+        if (predicateValue instanceof String) {
+            sqlClause = String.format("%s \"%s\"", clausePrefix, predicateValue);
+        } else if (predicateValue instanceof Byte ||
+                   predicateValue instanceof Integer ||
+                   predicateValue instanceof Long ||
+                   predicateValue instanceof Short ||
+                   predicateValue instanceof Double ||
+                   predicateValue instanceof Float) {
+            sqlClause = String.format("%s %s", clausePrefix, predicateValue);
+        }
+
+        return sqlClause;
+    }
```

Comment:
I think `addPredicateValueToSqlClause` is completing the statement that the objc code generates [here][1], that is it returns something like `= "mike"`. In that case, the `String.format` statements in this should have placeholders (`?`) for the `predicateValue` rather than using string substitution.

[1]: https://github.com/cloudant/CloudantQueryObjc/blob/master/Pod/Classes/CDTQQuerySqlTranslator.m#L390